mcs/tools/mono-api-html/MarkdownFormatter.cs

```diff
@@ -0,0 +1,199 @@
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+using System.Text;
+
+namespace Xamarin.ApiDiff {
+
+	public class MarkdownFormatter : Formatter {
+
+		public override string LesserThan => "<";
+		public override string GreaterThan => ">";
+
+		public override void BeginDocument (TextWriter output, string title)
+		{
+			output.WriteLine ($"# {title}");
+			output.WriteLine ();
+		}
+
+		public override void BeginAssembly (TextWriter output)
+		{
+			// this serves as ToC (table of content) entries so we skip the "Assembly: " prefix
+			output.WriteLine ($"## {State.Assembly}.dll");
+			output.WriteLine ();
+		}
+
+		public override void BeginNamespace (TextWriter output, string action)
+		{
+			output.WriteLine ($"### {action}Namespace {State.Namespace}");
+			output.WriteLine ();
+		}
+
+		public override void BeginTypeAddition (TextWriter output)
+		{
+			output.WriteLine ($"#### New Type: {@State.Namespace}.{State.Type}");
+			output.WriteLine ();
+			output.WriteLine ("```csharp");
+		}
+
+		public override void EndTypeAddition (TextWriter output)
+		{
+			output.WriteLine ("```");
+			output.WriteLine ();
+		}
+
+		public override void BeginTypeModification (TextWriter output)
+		{
+			output.WriteLine ($"#### Type Changed: {State.Namespace}.{State.Type}");
+			output.WriteLine ();
+		}
+
+		public override void BeginTypeRemoval (TextWriter output)
+		{
+			output.WriteLine ($"#### Removed Type {State.Namespace}.{State.Type}");
+		}
+
+		public override void BeginMemberAddition (TextWriter output, IEnumerable<XElement> list, MemberComparer member)
+		{
+			if (State.BaseType == "System.Enum") {
+				output.WriteLine ("Added value{0}:", list.Count () > 1 ? "s" : String.Empty);
+			} else {
+				output.WriteLine ("Added {0}:", list.Count () > 1 ? member.GroupName : member.ElementName);
+			}
+			output.WriteLine ();
+			output.WriteLine ("```csharp");
+		}
+
+		public override void AddMember (TextWriter output, MemberComparer member, bool isInterfaceBreakingChange, string obsolete, string description)
+		{
+			output.Write (obsolete);
```

Comment:
You should check mosts values for special characters (e.g. backtick) and escape them